pathogen/src/lib.rs

```diff
@@ -73,6 +82,10 @@ where
         Change::Update { key_path, value }.into()
     }
 
+    pub fn delete(key_path: KeyPath<Root, T>) -> ChangeOf<Root> {
+        Change::Delete { key_path }.into()
+    }
+
```

Comment:
I think this should only be implemented for `Change<Root, BTreeMap<K, T>>` (if we want to allow any key type, otherwise `K` shoud be `String`)

Pathogen should do its best to prevent generating invalid changes. It is tempting to say "delete is just a delete of keypath" but it should really be typechecked to say "the delete is targeting a dynamic collection which we can delete from". So the `Change::Delete` should probably carry a keypath to the map itself, and the key to be removed.

Reply:
Actually, on second pass, looking at how splice works, it's more a case of the `key_path` being `KeyPath<Root, BTreeMap<K, T>>` and the change carrying `key: K`. 

Sorry for the original slapdash suggestion.